### PR TITLE
add dimmer-configure-magit

### PR DESCRIPTION
**Pull request by tomkoelman**
I added ` *transient*` to `dimmer-buffer-exclusion-regexps`

### CODE REVIEW

**Comment by gonewest818** [comment]
Thank you!  Do you mind updating the README and CHANGELOG as well?

**Comment by tomkoelman** [comment]
> Thank you! Do you mind updating the README and CHANGELOG as well?

I just added a new commit with those changes. I didn't know what the bracketed number (e.g. [#37]) should be so I just left it out.

**Comment by gonewest818** [comment]
Thank you, totally fine. If there had been a reported issue it would be that number.  But it’s not worth opening a ticket just to close it again.

**Comment by tomkoelman** [comment]
Got it. I like the package, thanks for sharing.

File: README.md [hunk 1]
```diff
@@ -52,8 +52,8 @@ ensure org-mode buffers are not dimmed.
 * `dimmer-configure-posframe` is a convenience function for posframe
 users to ensure posframe buffers are not dimmed.
 
-* `dimmer-configure-which-key` is a convenience function for which-key
-users to ensure which-key popups are not dimmed.
```
Comment (gonewest818) [inline README.md]: It seems you accidentally deleted the documentation for dimmer-configure-which-key. 

And actually, I'm trying to keep these dimmer-configure-xxx functions in alphabetical order, so can this move up to the appropriate place in the list?  Same for the header comment in the source file.

File: README.md [hunk 2]
```diff
@@ -52,8 +52,8 @@ ensure org-mode buffers are not dimmed.
...
+* `dimmer-configure-magitis a convenience function for magit users to
+ensure transients are not dimmed.
```
Comment (gonewest818) [inline README.md]: typo: you're missing the closing quote and a space is needed after `dimmer-configure-magit`

File: dimmer.el [hunk 3]
```diff
@@ -264,6 +267,13 @@ uses a buffer name that fits this pattern."
     (add-to-list
      'dimmer-prevent-dimming-predicates #'which-key--popup-showing-p)))
 
+;;;###autoload
+(defun dimmer-configure-magit ()
+  "Convenience settings for magit users."
+  (with-no-warnings
```
Comment (gonewest818) [inline dimmer.el]: Technically we don't need a `with-no-warnings` here. There are cases where we need to refer to predicate functions in other packages and the bytecompiler will complain that it doesn't recognize the symbol. But here the regexp doesn't depend on symbols in the package.

**Review by gonewest818 — CHANGES_REQUESTED** [review]
A few minor changes noted in the comments. Thank you again for contributing!

**Comment by tomkoelman** [comment]
Thanks for the detailed feedback. I think I addressed every comment you made in the last three commits.

**Comment by gonewest818** [comment]
Merged!   Glad you like the package!

**Pull request merged**